### PR TITLE
Added ApplicationConfiguration to set generic configurations for application

### PR DESCRIPTION
**Pull request by JeanExtreme002**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by lesters — CHANGES_REQUESTED** [review]
Thanks for this! I feel however that this parameter should not be a top-level argument to the constructor, as this sets a certain configuration setting. There could be many such. What do you think about making this a bit more general, that can support any config setting?

**Comment by JeanExtreme002** [comment]
@lesters Done! The only thing I'll do now is fixing the `services_to_text` text fomatting, because for some reason `</config>` is going out from indentation. But everything is already working! :smile:

**Review by lesters — COMMENTED** [review]
Looks good, but there are test errors?

**Comment by JeanLoui-Escavador** [comment]
> Looks good, but there are test errors?

Other pull requests had that same problem on Screwdriver/7055/PR:unit-tests

File: vespa/package.py [hunk 1]
```diff
@@ -1370,6 +1370,48 @@ def __repr__(self):
...
+
+        :param name: Configuration name.
+        :param value: Either a string or a Dict (it may be a nested dict) of values.
+
+        Example:
+        >>> ApplicationConfiguration(
+        >>>     name="container.handler.observability.application-userdata",
+        >>>     value={"version": "my-version"}
+        >>> )
+        """
```
Comment (lesters) [inline vespa/package.py:1389]: Unit test fail is related to this. The doctesting fails here. Should be "..." on these lines?

**Review by lesters — CHANGES_REQUESTED** [review]
Looks ok, but please fix unit test failures.

**Comment by JeanExtreme002** [comment]
Sorry for my mistake. It's done! @lesters

**Review by lesters — APPROVED** [review]
LGTM!

**Pull request merged**